packages/amplify_core/lib/src/types/storage/upload_file_operation.dart

```diff
@@ -23,9 +23,7 @@ abstract class StorageUploadFileOperation<
         Request extends StorageUploadFileRequest,
         Result extends StorageUploadFileResult>
     extends StorageOperation<Request, Result>
-// TODO(HuiSF): implement controllable operation APIs when AWSHttpOperation
-//  is returned from S3Client APIs.
-/* implements StorageControllableOperation */ {
+    implements StorageControllableOperation {
```

Comment:
There's one on `uploadData` operation as well - will that be addressed separately?

Reply:
Good q, `uploadData` is not pause-able as s3 multipart upload is not applicable where file size is not determinable. 
We should expose at least the `cancel` API for upload data operation.

Reply:
Updated.